File: backend/api/models/empresa_model.py
from api.app import db

class EmpresaModel(db.Model):
    __tablename__ = 'empresa'
    idEmpresa = db.Column(db.Integer, autoincrement=True, primary_key=True, nullable=False)
    nomeEmpresa = db.Column(db.String(50), nullable=False)
    dtRefSistema = db.Column(db.String(7), nullable=False)
    dtImplantacao = db.Column(db.String(50), nullable=False)
    endereco = db.Column(db.String(255), nullable=False)
    cidade = db.Column(db.String(50), nullable=False)
    cnpj = db.Column(db.String(50), nullable=False)

File: Express/src/models/empresa_model.ts
import { DataTypes } from "sequelize";
import { db } from "../../db/config";

export const EmpresaModel = db.define('empresa', {
  id_empresa: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  nome_empresa: {
    type: DataTypes.STRING(100),
    allowNull: false,
  },
  dt_ref_sistema: {
    type: DataTypes.DATE,
    allowNull: false,
  },
  cep: {
    type: DataTypes.CHAR(8),
    allowNull: false,
  },
  num: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  cpf_cnpj: {
    type: DataTypes.CHAR(14),
    allowNull: false,
  },
}, {
  indexes: [
    {
      name: 'idx_nome',
      fields: ['nome_empresa'],
    },
    {
      name: 'idx_cpf_cnpj',
      fields: ['cpf_cnpj'],
    },
  ],
  timestamps: true,
  createdAt: 'dt_criacao',
  updatedAt: 'dt_atualizacao',
});
